test/modules/DirectMethodReceiver/src/DirectMethodReceiver.cs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace DirectMethodReceiver
+{
+    using System;
+    using System.Net;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.Extensions.Logging;
+    class DirectMethodReceiver : IDisposable
+    {
+        Guid batchId;
+        IConfiguration configuration;
+        long directMethodCount = 1;
+        ILogger logger;
+        ModuleClient moduleClient;
+        TestResultReportingClient testResultReportingClient;
+
+        public DirectMethodReceiver(
+            ILogger logger,
+            IConfiguration configuration,
+            ModuleClient moduleClient,
+            TestResultReportingClient testResultReportingClient)
+        {
+            Preconditions.CheckNotNull(logger, nameof(logger));
+            Preconditions.CheckNotNull(configuration, nameof(configuration));
+            Preconditions.CheckNotNull(moduleClient, nameof(moduleClient));
+
+            this.logger = logger;
+            this.configuration = configuration;
+            this.moduleClient = moduleClient;
+            this.testResultReportingClient = testResultReportingClient;
+            this.batchId = Guid.NewGuid();
+        }
+
+        public static async Task<DirectMethodReceiver> CreateAsync(
+            ILogger logger,
+            IConfiguration configuration)
+        {
+            Preconditions.CheckNotNull(logger, nameof(logger));
+            Preconditions.CheckNotNull(configuration, nameof(configuration));
+
+            ModuleClient moduleClient = await ModuleUtil.CreateModuleClientAsync(
```

Comment:
Move module client creation inside constructor

Reply:
Can a constructor be async?